presto-main/src/main/java/com/facebook/presto/operator/scalar/RowIndeterminateOperator.java

```diff
@@ -0,0 +1,184 @@
+import static com.facebook.presto.operator.scalar.ScalarFunctionImplementation.NullConvention.USE_NULL_FLAG;
+import static com.facebook.presto.spi.function.OperatorType.INDETERMINATE;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.sql.gen.InvokeFunctionBytecodeExpression.invokeFunction;
+import static com.facebook.presto.sql.gen.SqlTypeBytecodeExpression.constantType;
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static com.facebook.presto.util.CompilerUtils.defineClass;
+import static com.facebook.presto.util.CompilerUtils.makeClassName;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.bytecode.Access.FINAL;
+import static io.airlift.bytecode.Access.PUBLIC;
+import static io.airlift.bytecode.Access.STATIC;
+import static io.airlift.bytecode.Access.a;
+import static io.airlift.bytecode.Parameter.arg;
+import static io.airlift.bytecode.ParameterizedType.type;
+import static io.airlift.bytecode.expression.BytecodeExpressions.constantFalse;
+import static io.airlift.bytecode.expression.BytecodeExpressions.constantInt;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RowIndeterminateOperator
+        extends SqlOperator
+{
+    public static final RowIndeterminateOperator ROW_INDETERMINATE = new RowIndeterminateOperator();
+
+    private RowIndeterminateOperator()
+    {
+        super(INDETERMINATE, ImmutableList.of(withVariadicBound("T", "row")), ImmutableList.of(), BOOLEAN.getTypeSignature(), ImmutableList.of(parseTypeSignature("T")));
+    }
+
+    @Override
+    public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        checkArgument(arity == 1, "Expected arity to be 1");
+        Type type = boundVariables.getTypeVariable("T");
+        Class<?> indeterminateOperatorClass = generateIndeterminate(type, functionRegistry);
+        MethodHandle methodHandle = methodHandle(indeterminateOperatorClass, "indeterminate", type.getJavaType(), boolean.class);
+        return new ScalarFunctionImplementation(
+                false,
+                ImmutableList.of(
+                        valueTypeArgumentProperty(USE_NULL_FLAG)),
+                methodHandle,
+                isDeterministic());
+    }
+
+    private static Class<?> generateIndeterminate(Type type, FunctionRegistry functionRegistry)
+    {
+        CallSiteBinder binder = new CallSiteBinder();
+
+        // Embed the MD5 hash code of input and output types into the generated class name instead of the raw type names
+        // to avoid hitting class naming length or character set limits
+        byte[] md5Suffix = Hashing.goodFastHash(128).hashBytes(type.toString().getBytes(UTF_8)).asBytes();
```

Comment:
I had a offline discussion with @haozhun on whether we need to add hashed type name as class name suffix.

For bytecode generated parametric scalar functions, Presto adds the types as suffix, such as `ArrayCast$Bigint$Varchar` to help debugging. Note this is not to avoid class name collisions, since `CompilerUtils::makeClassName` will append a unique `CLASS_ID` at the end. 

However, for generated functions over row types, appending raw type names might make class name hitting length limitation. Thus in `RowToRowCast` md5 suffix is introduced to avoid this.

Yet we find this defeats the original purpose of appending type names (to help debug), since now the class name is anyway not human-readable. The only usage we can think is, one could still **verify** the type of a generated function by write down the type names and compute the hash. Such verification doesn't seem to be useful, given the cost and potential benefit. 

Given this, I suggest to remove the `md5Suffix` and just use `RowIndeterminateOperator` as the base class name. In the future, we could add a truncated type name to help debug, but this is beyond the scope of this PR.